src/kOS.Safe/Encapsulation/ListValue.cs

```diff
@@ -202,74 +250,42 @@ private string ListDump()
 
             if (truncateHere)
             {
-                --currentNestDepth;                
+                --currentNestDepth;
                 return TerseDump();
             }
             var contents = new StringBuilder();
-            contents.AppendLine( TerseDump() + ":" );
-            var indent = new string(' ', currentNestDepth*SPACES_PER_INDENT);
-            for (int i = 0 ; i < list.Count ; ++i)
+            contents.AppendLine(TerseDump() + ":");
+            var indent = new string(' ', currentNestDepth * SPACES_PER_INDENT);
+            for (int i = 0; i < internalList.Count; ++i)
             {
-                contents.AppendLine( string.Format("{0}[{1,2}]= {2}", indent, i, list[i]) );
+                contents.AppendLine(string.Format("{0}[{1,2}]= {2}", indent, i, internalList[i]));
             }
-            --currentNestDepth;                
+            --currentNestDepth;
             return contents.ToString();
         }
-        public static ListValue CreateList<T>(IEnumerable<T> sourceList)
-        {
-            return new ListValue(sourceList.Cast<object>().ToList());
-        }
-
-        #region IIndexable Members
-
-        public object GetIndex(int index)
-        {
-            return list[index];
-        }
-
-        public void SetIndex(int index, object value)
-        {
-            list[index] = value;
-        }
-
-        #endregion IIndexable Members
     }
 
-    public class Enumerator : Structure
+    public class ListValue : ListValue<object>
```

Comment:
Uhhh... so you're allowed to have two entirely different classes that share the same identifier name but differ only in the fact that one is templated with <> and the other is not?  Ugh.  that's ugly.